master/internal/stream/system_intg_test.go

```diff
@@ -0,0 +1,200 @@
+//go:build integration
+// +build integration
+
+package stream
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/require"
+
+	detContext "github.com/determined-ai/determined/master/internal/context"
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/stream"
+	"github.com/determined-ai/determined/master/test/streamdata"
+)
+
+func simpleUpsert(i stream.PreparableMessage) interface{} {
+	return i
+}
+
+func recordDeletion(s1, s2 string) interface{} {
+	fmt.Println("recording deletion:", s1+" and "+s2)
+	return s2
+}
+
+type startupReadWriter struct {
+	data           []interface{}
+	startupMessage *StartupMsg
+}
+
+func (t *startupReadWriter) ReadJSON(data interface{}) error {
```

Comment:
please change the pointer receiver to not be called `t`, which overlaps with the canonical name for a `*testing.T` object, so it makes the code really confusing